Short-Answer/Algorithms_Answers.md

```diff
@@ -1 +1,64 @@
-Add your answers to the Algorithms exercises here.
\ No newline at end of file
+Add your answers to the Algorithms exercises here.
+
+
+A ) the code would never end. the value of n would accelerate away from
+"a" linearly i believe. the while loop could be considered n^3 and the inner loop 
+its n^2+a which if subtracted reaches O(n) now I also have to consider "a" so maybe the time complexity is O(n + a)?  
+
+
+B) I think the time complexity of this function is O(n^4) 
+a normal double for loop is o(n^2) using this I cam to my conclusion 
+
+for i in range n,   length of n  
+        increase i 
+        for j in range i + 1, to n    
+        <!-- range is still determined by n but could be getting smaller because they increase the value of i by one -->
+            add one to j
+            for k in range k+ 1, to n      
+            <!-- range n is still being approached but this time k times k in being incrase by one for some reason -->
+                for l in range k + 1 to k + 10     
+                <!-- l is limited by K instead of n, k has a hard limit of k+10 instead of n times              -->
+                    increase l by one             
+                    <!-- loop runs until until k runs out then incrases sum by one. -->
+                    increase sum by 1
+
+ 
+
+
+C) bunnies looks to be a number of bunnies. if no bunnies exist return 0
+if not then return double the amount of bunnes inputed 
+2 + O(n-1) time complexity?
```

Comment:
Close! This would be equivalent to a loop from `0` to `bunnies` so it would be `O(n)`.

Think about it as if it was a while loop. What we are saying is if bunnies == 0 then return, if not add 2 to the number of bunnies we have, and do it again with one less bunny